### PR TITLE
Fix concurrency write map panic

### PR DESCRIPTION
**Pull request by winwill2012**
**Fix concurrency write map panic:**

fatal error: concurrent map read and map write

goroutine 163 [running]:
runtime.throw({0x109be48, 0xf5ae00})
        /usr/local/go/src/runtime/panic.go:1198 +0x71 fp=0xc0041e0630 sp=0xc0041e0600 pc=0x435bf1
runtime.mapaccess2_faststr(0xc003c2e5e8, 0x26, {0x107e883, 0x7})
        /usr/local/go/src/runtime/map_faststr.go:116 +0x3d4 fp=0xc0041e0698 sp=0xc0041e0630 pc=0x412db4
github.com/zhenghaoz/gorse/worker.(*Worker).exploreRecommend(0xc00025e000, {0xc00bad2c00, 0x28, 0x28}, 0x1089476, {0xc000f2ba00, 0xf})
        /Users/mac/code/gorse/worker/worker.go:713 +0x90 fp=0xc0041e07a0 sp=0xc0041e0698 pc=0xe68350
github.com/zhenghaoz/gorse/worker.(*Worker).Recommend.func2(0x0, 0x7ffb)
        /Users/mac/code/gorse/worker/worker.go:602 +0x2e35 fp=0xc0041e1f48 sp=0xc0041e07a0 pc=0xe65c95
github.com/zhenghaoz/gorse/base.Parallel.func2(0x0)
        /Users/mac/code/gorse/base/parallel.go:67 +0xc4 fp=0xc0041e1fc8 sp=0xc0041e1f48 pc=0x78e824
github.com/zhenghaoz/gorse/base.Parallel·dwrap·1()
        /Users/mac/code/gorse/base/parallel.go:72 +0x2d fp=0xc0041e1fe0 sp=0xc0041e1fc8 pc=0x78e72d
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc0041e1fe8 sp=0xc0041e1fe0 pc=0x468401
created by github.com/zhenghaoz/gorse/base.Parallel
        /Users/mac/code/gorse/base/parallel.go:57 +0x130

### CODE REVIEW

**Comment by zhangzhenghao** [comment]
LGTM. Thanks for your contribution 😁

File: config/config.go [hunk 1]
```diff
@@ -126,14 +127,25 @@ type RecommendConfig struct {
...
 	ExploreRecommend             map[string]float64 `toml:"explore_recommend"`
-	ItemNeighborType             string             `toml:"item_neighbor_type"`
-	UserNeighborType             string             `toml:"user_neighbor_type"`
-	EnableLatestRecommend        bool               `toml:"enable_latest_recommend"`
-	EnablePopularRecommend       bool               `toml:"enable_popular_recommend"`
-	EnableUserBasedRecommend     bool               `toml:"enable_user_based_recommend"`
-	EnableItemBasedRecommend     bool               `toml:"enable_item_based_recommend"`
-	EnableColRecommend           bool               `toml:"enable_collaborative_recommend"`
-	EnableClickThroughPrediction bool               `toml:"enable_click_through_prediction"`
+	ExploreRecommendLock         sync.Mutex
```
Comment (zhangzhenghao) [inline config/config.go]: Lowercase is better to avoid copy.

File: config/config_test.go [hunk 2]
```diff
@@ -56,6 +56,14 @@ func TestLoadConfig(t *testing.T) {
 	assert.Equal(t, 1, config.Recommend.RefreshRecommendPeriod)
 	assert.Equal(t, []string{"item_based", "latest"}, config.Recommend.FallbackRecommend)
 	assert.Equal(t, map[string]float64{"popular": 0.1, "latest": 0.2}, config.Recommend.ExploreRecommend)
+	value, exist := config.Recommend.GetExploreRecommend("popular")
+	assert.Equal(t, true, exist)
+	assert.Equal(t, 0.1, value)
+	value, exist = config.Recommend.GetExploreRecommend("latest")
+	assert.Equal(t, true, exist)
+	assert.Equal(t, 0.2, value)
+	value, exist = config.Recommend.GetExploreRecommend("unknown")
```
Comment (zhangzhenghao) [inline config/config_test.go]: This value could be checked or just ignore it by using _.

**Review by zhangzhenghao — CHANGES_REQUESTED** [review]

File: config/config_test.go (hunk 2 above)
Comment (winwill2012) [reply config/config_test.go]: ok

File: config/config.go (hunk 1 above)
Comment (winwill2012) [reply config/config.go]: ok~

File: config/config.go [hunk 3]
```diff
@@ -134,6 +135,17 @@ type RecommendConfig struct {
 	EnableItemBasedRecommend     bool               `toml:"enable_item_based_recommend"`
 	EnableColRecommend           bool               `toml:"enable_collaborative_recommend"`
 	EnableClickThroughPrediction bool               `toml:"enable_click_through_prediction"`
+	exploreRecommendLock         *sync.Mutex
```
Comment (zhangzhenghao) [inline config/config.go]: Pointer break the UT😂

**Review by zhangzhenghao — CHANGES_REQUESTED** [review]

**Review by zhangzhenghao — APPROVED** [review]

**Pull request merged**